syntax = "proto3";

package grpc.health.v1;

message HealthCheckRequest {
    string service = 1;
}

message HealthCheckResponse {
    enum ServiceStatus {
        UNKOWN = 0;
        SERVING = 1;
        NOT_SERVING = 2;
        SERVICE_UNKOWN = 3;
    }
    ServiceStatus status = 1;
}

service Health {
    rpc Check(HealthCheckRequest) returns (HealthCheckResponse);
    rpc Watch(HealthCheckRequest) returns (stream HealthCheckResponse);
}